src/conduits/RwaMultiSwapOutputConduit.t.sol

```diff
@@ -0,0 +1,790 @@
+        conduit.clap(address(psm));
+
+        vm.expectEmit(true, false, false, false);
+        emit Hook(address(psm));
+        conduit.hook(address(psm));
+        assertEq(conduit.psm(), address(psm));
+        assertEq(conduit.gem(), address(usdx));
+
+        // pick zero address
+        vm.expectEmit(true, false, false, false);
+        emit Hook(address(0));
+        conduit.hook(address(0));
+        assertEq(conduit.psm(), address(0));
+        assertEq(conduit.gem(), address(0));
+    }
+
+    function testPush() public {
+        assertEq(outputConduit.to(), me);
+        assertEq(outputConduit.psm(), address(psm));
+        assertEq(outputConduit.gem(), address(usdx));
+        assertEq(usdx.balanceOf(me), 0);
+        assertEq(usdx.balanceOf(address(outputConduit)), 0);
+        assertEq(dai.balanceOf(address(me)), 1_000 * WAD);
+
+        dai.transfer(address(outputConduit), 500 * WAD);
+
+        assertEq(dai.balanceOf(me), 500 * WAD);
+        assertEq(dai.balanceOf(address(outputConduit)), 500 * WAD);
+
+        vm.expectEmit(true, true, false, false);
+        emit Push(address(psm), address(me), 500 * USDX_BASE_UNIT);
+        outputConduit.push();
+
+        assertEq(usdx.balanceOf(address(me)), 500 * USDX_BASE_UNIT);
+        assertApproxEqAbs(dai.balanceOf(address(outputConduit)), 0, USDX_DAI_CONVERSION_FACTOR);
+        assertEq(outputConduit.to(), address(0));
+        assertEq(outputConduit.gem(), address(0));
+        assertEq(outputConduit.psm(), address(0));
+    }
+
+    function testPushAfterChangingPsm() public {
+        // Init new PSM
+        usdx.mint(USDX_MINT_AMOUNT);
+
+        AuthGemJoin5 join = new AuthGemJoin5(address(vat), ilk, address(usdx));
+        vat.rely(address(join));
+        DssPsm newPsm = new DssPsm(address(join), address(daiJoin), address(vow));
+        join.rely(address(newPsm));
+        join.deny(me);
+
+        usdx.approve(address(join));
+        newPsm.sellGem(me, USDX_MINT_AMOUNT);
+
+        // Change PSM
+        outputConduit.clap(address(newPsm));
+        outputConduit.hook(address(newPsm));
+
+        assertEq(outputConduit.to(), me);
+        assertEq(outputConduit.psm(), address(newPsm));
+        assertEq(usdx.balanceOf(me), 0);
+        assertEq(usdx.balanceOf(address(outputConduit)), 0);
+        assertEq(dai.balanceOf(address(me)), 2_000 * WAD);
+
+        dai.transfer(address(outputConduit), 500 * WAD);
+
+        assertEq(dai.balanceOf(me), 1_500 * WAD);
+        assertEq(dai.balanceOf(address(outputConduit)), 500 * WAD);
+
+        vm.expectEmit(true, true, false, false);
+        emit Push(address(newPsm), address(me), 500 * USDX_BASE_UNIT);
+        outputConduit.push();
+
+        assertEq(usdx.balanceOf(address(me)), 500 * USDX_BASE_UNIT);
+        assertApproxEqAbs(dai.balanceOf(address(outputConduit)), 0, USDX_DAI_CONVERSION_FACTOR);
+        assertEq(outputConduit.to(), address(0));
+        assertEq(outputConduit.psm(), address(0));
+    }
```

Comment:
Maybe we should make this test use a PSM with a new token with different `decimals()`.